### PR TITLE
Microsoft.Data.Sqlite: Address BLOB I/O API review feedback

### PR DESCRIPTION
**Pull request by bricelam**
cc @AlexanderTaeschner

### CODE REVIEW

File: test/Microsoft.Data.Sqlite.Tests/SqliteBlobTest.cs [hunk 1]
```diff
@@ -416,15 +416,15 @@ public void Write_throws_when_not_writable()
...
+            var stream = CreateStream(readOnly: false);
             stream.Dispose();
 
             var ex = Assert.Throws<ObjectDisposedException>(
                 () => stream.Write(new byte[] { 3 }, 0, 1));
         }
 
-        protected Stream CreateStream(bool writable = false)
-            => new SqliteBlob(_connection, Table, Column, Rowid, writable);
+        protected Stream CreateStream(bool readOnly = false)
```
Comment (AlexanderTaeschner) [inline test/Microsoft.Data.Sqlite.Tests/SqliteBlobTest.cs]: Should'nt this be changed to `bool readOnly = true`?

Comment (bricelam) [reply test/Microsoft.Data.Sqlite.Tests/SqliteBlobTest.cs]: I wanted to match the product. For most of the tests the value doesn’t matter. I made the value explicit where it mattered.

**Comment by bricelam** [comment]
@dotnet-bot test OSX 10.12 Release Build please

**Comment by bricelam** [comment]
Failed   Microsoft.Data.Sqlite.SqliteConnectionTest.CreateFunction_works_when_null
    Error Message:
     System.InvalidOperationException : Handle is not initialized.
    Stack Trace:
       at System.Runtime.InteropServices.GCHandle.ThrowInvalidOperationException_HandleIsNotInitialized()
       at System.Runtime.InteropServices.GCHandle.Free()
       at SQLitePCL.scalar_function_hook_info.free()
       at SQLitePCL.SQLite3Provider_e_sqlite3.my_sqlite3_create_function(IntPtr db, String name, Int32 nargs, Int32 flags, Object v, delegate_function_scalar func)
       at SQLitePCL.SQLite3Provider_e_sqlite3.SQLitePCL.ISQLite3Provider.sqlite3_create_function(IntPtr db, String name, Int32 nargs, Int32 flags, Object v, delegate_function_scalar func)
       at SQLitePCL.raw.sqlite3_create_function(sqlite3 db, String name, Int32 nArg, Int32 flags, Object v, delegate_function_scalar func)
       at Microsoft.Data.Sqlite.SqliteConnection.CreateFunctionCore[TState,TResult](String name, Int32 arity, TState state, Func`3 function, Boolean isDeterministic) in /_/src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs:line 498
       at Microsoft.Data.Sqlite.SqliteConnection.CreateFunction[TResult](String name, Func`1 function, Boolean isDeterministic) in /_/src/Microsoft.Data.Sqlite.Core/SqliteConnection.CreateFunction.cs:line 18
       at Microsoft.Data.Sqlite.SqliteConnectionTest.CreateFunction_works_when_null() in /_/test/Microsoft.Data.Sqlite.Tests/SqliteConnectionTest.cs:line 583

**Pull request merged**